src/Promitor.Scraper.Tests.Unit/Serialization/v1/V1SerializationTests.cs

```diff
@@ -180,7 +180,7 @@ public void Deserialize_SerializedYaml_CanDeserializeToRuntimeModel()
             var yaml = _configurationSerializer.Serialize(_metricsDeclaration);
 
             // Act
-            var runtimeModel = _configurationSerializer.Deserialize(yaml);
+            var runtimeModel = _configurationSerializer.Deserialize(yaml, new ErrorReporter());
```

Comment:
It's not optional, and I wouldn't really want to make it optional just for the sake of this test. If there were places in the actual runtime code that were calling the method without a reporter I'd say it made sense to make it optional, but since there's only one usage, in the MetricsDeclarationProvider, and it always passes in a reporter, I think it's ok to leave it as it is.

I've realised that there was already an _errorReporter field defined in this test class, so I've updated this test to use it, and I've also added a Guard.NotNull() to ConfigurationSerializer.Deserialize() since I've realised there was a guard against the yaml parameter, but not against the reporter.